### ISSUE TITLE
Add rolling-origin forecast model selection script

### ISSUE THREAD

**Issue by jesuscontrerasocana** [jesuscontrerasocana/scaffold#34]
## Goal

Add a standalone script that performs a formal, reproducible selection of:

1. forecast model architecture; and
2. training lookback period.

The script should reuse the existing forecasting and evaluation code rather than duplicate feature construction, prediction, or metric logic.

This is a **forecast model-selection tool only**. It should not run the battery optimizer.

## Candidate configurations

Evaluate the Cartesian product of:

### Models

- `ridge`
- `ridge_decomposed`
- `ridge_hgbr_decomposed`

Use the existing `Forecaster(..., model_name=...)` implementations. Do not introduce separate model implementations in the selection script.

### Training lookbacks

- `1m`: previous 1 calendar month
- `3m`: previous 3 calendar months
- `expanding`: all available observations before the validation period

The lookback is part of the model configuration and must be selected jointly with the model architecture.

## Validation method

Use **rolling-origin monthly validation**.

Default development folds:

```text
Train through 2026-03-31 -> validate April 2026
Train through 2026-04-30 -> validate May 2026
Train through 2026-05-31 -> validate June 2026
```

The actual training start depends on the lookback.

Example for `3m`:

```text
Jan-Mar -> April
Feb-Apr -> May
Mar-May -> June
```

Example for `expanding`:

```text
all available data through Mar -> April
all available data through Apr -> May
all available data through May -> June
```

For each fold:

1. determine the validation month;
2. select only data strictly before the validation start for model fitting;
3. apply the configured lookback to that training data;
4. instantiate the requested existing `Forecaster`;
5. fit it once;
6. roll forecasts through the validation month using the same information boundary as the existing forecast evaluation script;
7. calculate fold metrics;
8. record forecast inference runtime.

During validation, realized observations earlier in the validation month may become forecast history exactly as they do in the existing rolling evaluation/harness. They must **not** be used to refit the model within the fold.

## Reuse existing code

Build this as a thin orchestration layer around the current implementation.

Reuse from `scripts/evaluate_forecast.py` where appropriate:

- `collect_forecasts()`
- `calculate_metrics()`
- `component_metrics()`
- `lead_metrics()`

Reuse:

- `pipeline.data.load_timeseries`
- `pipeline.forecaster.Forecaster`
- `pipeline.harness.RunConfig`
- `pipeline.specs.SiteSpecs`

Do not copy the forecasting loop or reimplement the harness information boundary in the new script.

If small helpers currently live inside `evaluate_forecast.py` and need to be reused cleanly, a small refactor is fine. Avoid creating a broader evaluation framework.

## New script

Add something like:

```text
scripts/select_forecast_model.py
```

Example usage:

```bash
uv run python scripts/select_forecast_model.py \
    --data data/history.csv \
    --from 2026-04-01 \
    --to 2026-06-30 \
    --models ridge ridge_decomposed ridge_hgbr_decomposed \
    --lookbacks 1m 3m expanding \
    --out out/model_selection.csv
```

The April-June range defines the monthly validation folds.

Keep reasonable defaults for this assignment so the standard experiment can be run without many arguments.

## Metrics

For every `(model, lookback, validation_month)` fold record at least:

### Net load

- MAE
- RMSE
- bias
- nMAE

### Components, when available

- load MAE
- PV MAE
- PV RMSE

### Horizon-specific net-load accuracy

Report:

- lead 1 MAE
- first-hour MAE: leads 1-4 pooled together
- first-4-hours MAE: leads 1-16 pooled together
- full-horizon MAE

Use the underlying forecast comparison rows when computing pooled horizon metrics rather than averaging per-lead MAEs.

### Runtime

Record at least:

- forecast inference seconds for the fold
- number of forecast decisions
- forecast seconds per decision

Training time may also be recorded separately because HGBR training can be materially slower, but training runtime should not be part of the primary ranking.

## Aggregate results

Produce one row per `(model, lookback)` configuration containing metrics aggregated across validation folds.

Also retain the per-fold results so stability across months is visible.

Suggested outputs:

```text
out/model_selection.csv
out/model_selection_folds.csv
```

`model_selection.csv` should be easy to sort/read and contain one row per candidate configuration.

Do not require persisted model artifacts for every validation fold unless needed for the existing interface. These are temporary model-selection fits and can remain in memory.

## Selection rule

Use a simple deterministic selection rule rather than an arbitrary weighted score.

### Primary criterion

Lowest **mean validation net-load MAE** across the monthly folds.

### Tie-breaking / diagnostic criteria

If configurations are very close, inspect in this order:

1. lead-1 / first-hour MAE;
2. RMSE;
3. absolute bias;
4. stability across validation months;
5. inference runtime.

Runtime is primarily a feasibility constraint rather than something that should outweigh meaningful forecast-quality improvements.

Do not build a complex composite score.

The script should clearly print the selected configuration, for example:

```text
Selected model: ridge_hgbr_decomposed
Selected lookback: 3m
Mean validation net MAE: ...
Lead-1 MAE: ...
Forecast seconds/decision: ...
```

## Final-training guidance

The model-selection script should **not evaluate August**.

August remains the untouched final holdout.

Once a `(model, lookback)` configuration is selected, final training should use that same lookback rule with the latest allowed training cutoff.

For example, if the winner is:

```text
ridge_hgbr_decomposed + 3m
```

then final training through July 31 should use the corresponding most recent 3-month training window rather than silently switching to expanding history.

It is sufficient for this issue to report the selected configuration; automatic production retraining is not required.

## Tests

Add focused tests for the orchestration logic, not expensive full-month model-selection runs.

Cover at least:

- `1m` lookback returns only the previous calendar month;
- `3m` lookback returns the correct three previous calendar months;
- `expanding` uses all available data strictly before validation start;
- validation data is never included in model fitting;
- monthly folds are generated correctly;
- the configured model name is passed to the existing `Forecaster`;
- first-hour metrics use leads 1-4;
- first-4-hour metrics use leads 1-16;
- aggregate metrics are computed across folds;
- deterministic selection chooses the lowest mean net-load MAE;
- tie-breaking is deterministic;
- output contains both per-fold and aggregate results.

Keep tests small by mocking/stubbing expensive model fitting where appropriate.

## Out of scope

Do not:

- tune Ridge or HGBR hyperparameters;
- add new forecast features;
- add new model families;
- run optimizer/battery simulations;
- use August for model selection;
- retrain models continuously within a validation month;
- change the production `train` / `simulate` CLI interface;
- change the behavior of the existing forecasters.

## Acceptance criteria

- A standalone model-selection script exists.
- Existing `Forecaster` implementations are reused.
- Existing forecast evaluation logic is reused rather than duplicated.
- `ridge`, `ridge_decomposed`, and `ridge_hgbr_decomposed` can be compared.
- `1m`, `3m`, and expanding training windows can be compared.
- Rolling-origin monthly validation is leakage-safe.
- April, May, and June can be used as the default development folds.
- Per-fold and aggregate metrics are written to CSV.
- Net MAE, short-horizon accuracy, bias/RMSE, and inference runtime are reported.
- The selected model + lookback is printed clearly.
- August is untouched.
- Existing tests continue to pass.
